### ISSUE TITLE
Using invalid github token ends up with wrong message while `POST` is used.

### ISSUE THREAD

**Issue by rifel123** [VirtusLab/git-machete#305]
Since my previous access token has expired, i face strange error message from git-machete:

<img width="1154" alt="Screenshot 2021-10-22 at 11 44 18" src="https://user-images.githubusercontent.com/73847352/138432813-0feab646-151e-436a-8bfd-afdf103ace6a.png">

Message is a little bit misleading, cause given endpoint exists,  and in public repositories there is always possibility to perform `GET` on it. For some reason while firing `POST` with invalid token, github API returns 404(not found) but i'd rather expect 401(not authorized) in such a case.

**Comment by BartoszSiwinski** [comment]
I'm not sure what are the expectations in this issue.

We can't control github API, so they do what they do. We could add a workflow that in case that response from github is 404, we try to send GET and if it passes through than we assume that it's because of authorization issue and raise 401. That being said it's kind of not responsibility of git-machete tool. It's github that has such issue from what I understand.

**Comment by rifel123** [comment]
Yes, github API acts weird in such a case. 

In public repositories you can perform GET on `pulls` endpoint without valid token and get the response of 200, but i am not sure what happens in private repos under same conditions. If both private and public repos acts similarly, then proposed solution sounds good. TBD: Can we assume that POST with status code 404 means `not authorized`?  Personally i think checking with GET is more precise, as there is still chance that we really hit endpoint which doesn't exists.

Basically scope of this task would to be tell user there is something wrong with authorization, currently `endpoint not found` message is not true.

**Comment by BartoszSiwinski** [comment]
Ok, but for now I'm trying to figure out if git-machete is even appropriate place to fix this. Shouldn't we add it as bug/issue to github repo or smth instead and call it a day? Otherwise we will be handling it only for this particular project and not for any other project that is using github + if github will patch it/change it we will have to adjust.

**Comment by PawelLipski** [comment]
Hmmm I doubt GitHub would ever fix this, particularly that they mostly now develop GraphQL (v4) rather than REST API (v3) AFAIK... anyway, as @rifel123 suggested, let's just aim at reducing the confusion in our users if they ever encounter such issue (quite likely now that GitHub encourages everybody to use tokens with limited lifespan, as opposed to the old never-expiring ones). So:

> Can we assume that POST with status code 404 means not authorized? 

Actually, can we **always** assume that 404 _might_ mean that the user is not authorized (not just for POST)? As in, can we have a generic 404 error message saying sth like _the repo doesn't exist, or you don't have the permission to access (expired token?)_ ❓

**Comment by BartoszSiwinski** [comment]
Hm... Out of your proposals I think that going with the most basic solution proposed by @PawelLipski might be the most reasonable. I'm not a fan of addressing issues of 3rd party libs inside local projects, because it can easily become maintenance overhead. Since the decision is that we want to help our user a little I would stop at expanding the error message to what we know can happen. If we go further into investigating of what exactly is happening we may become github diagnostic tool instead of git enhancement.

I will prepare PR with error enhancement.

**Comment by PawelLipski** [comment]
> we may become github diagnostic tool instead of git enhancement

Yeah, https://en.wikipedia.org/wiki/Inner-platform_effect is a very real threat all the time
